### ISSUE TITLE
Premium block flows: add E2E tests

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#34404]
Premium block flows (see https://github.com/Automattic/jetpack/pull/12823) should have an E2E test covering flow for a simple site via Calypso's block editor:

1. when having a site with a free plan
1. I can insert a simple payments block in the block editor
1. block can be saved and it doesn't show up in the site's frontend
1. in the editor block has "upgrade nudge" over it
1. I can purchase a plan by pressing the upgrade button
1. flow brings me back to the same editor page in Calypso

#### Considerations

- Code for this flow is in Jetpack but gated with `SHOW_BLOCK_UPGRADE_NUDGE` and `IS_WPCOM` constants.
- The code is synced to .com and is meant to be visible only for simple sites via Calypso.
- This might change in future and the flows should work also for Jetpack plugin

#### Should these tests run as part of:

- E2E suite in Calypso repo?
- Jetpack E2E suite in Calypso repo?
- E2E suite in Jetpack repo? (p9dueE-Ju-p2)

@brbrr thoughts or questions? :-)

**Comment by brbrr** [comment]
Interesting. Personally, I'd prefer puppeteer over selenium tests. (i.e. e2e's from Jetpack repo) just because it's so much comfortable to use.

As per my understanding, most of the functionality will be in Jetpack, so it make sense to run these tests in Jetpack. since we control the WP site, we can easily manipulate with its constants, etc. 

> (Jetpack) E2E suite in Calypso repo
You mean the existing calypso e2e tests, right? This one will work too. These tests have a rich infrastructure already, especially related to the iframed editor. But it's not clear how to test new changes with this approach.

**Comment by simison** [comment]
>I'd prefer puppeteer over selenium tests. (i.e. e2e's from Jetpack repo) just because it's so much comfortable to use.

I agree! Code changes in Jetpack repo are most likely going to break these flows so I'd lean towards having tests where the code is.

The thing is that the flow will look slightly different when in .com/Calypso iframe vs Jetpack. Maybe we could just add _some_ tests in Calypso E2E for those bits and otherwise have tests in Jetpack repo together with the code. I could imagine changes in iframe integration or Calypso checkout flow could break premium blocks flow, so covering both ends seems important.

So test flows would look like this, lemme know if I'm missing anything:

**In Jetpack E2E:**
1. Confirm that upgrade nudge and paid blocks are not available for free sites when constant(s) are disabled
1. Enable constant(s)
1. when having a site with a free plan
1. I can insert a simple payments block in the block editor
1. block can be saved and it doesn't show up in the site's frontend
1. in the editor block has "upgrade nudge" over it
1. I can purchase a plan by pressing the upgrade button
1. Confirm that flow brings me back to the same editor page

**In Calypso E2E suite for simple-sites**
1. Open block editor for a free site
1. Confirm that pressing "upgrade" escapes the iframe
1. Confirm that after checkout, customer lands back in the correct URL in Calypso

Can we rest assured that running these tests won't affect any tracks events collected from these flows? Both Jetpack/Jurassic ninja & simple-sites Calypso iframe -scenarios.

Noting that Jetpack E2E tests are disabled at the moment. (ref p9dueE-Ju-p2#comment-1136)

**Comment by brbrr** [comment]
seems like a solid plan! 

One thing is to keep in mind that Jetpack E2E tests buying a Pro plan during setup (for premium blocks). We use a sandbox cookie for that, meaning when the cookie is set - sandbox plan data would be pulled. If we remove that cookie - production data would be used. It's a bit hacky but should work in your case. https://github.com/Automattic/jetpack/blob/master/tests/e2e/lib/pages/page.js#L60

**Comment by xpurichan** [comment]
I have tested the behaviour for adding the Pay with Paypal block (formerly called Simple Payments) to a WordPress.com website with a free plan. I am able to add the block, but I get a prompt stating "Upgrade your plan to use this premium block" as seen here:

![Screen Shot 2021-02-08 at 5 16 46 PM](https://user-images.githubusercontent.com/33190975/107288495-27de1000-6a29-11eb-8327-a73b14b5362f.jpg)

Seems as this idea has been implemented. Feel free to reopen if I'm mistaken!

**Comment by brbrr** [comment]
To be fair, I don't think "Fixed" status fits into this issue. It isn't a bug, but rather an enhancement for E2E test coverage.

**Comment by simison** [comment]
@apeatling I'll put this on View backlog, but feel free to close if E2E coverage is already sufficient. Thx!
